src/current-user.js

```diff
@@ -670,59 +670,112 @@ export class CurrentUser {
   }
 
   enablePushNotifications(config = {}) {
+    try {
+      const {
+        onClick: onClickCallback,
+        serviceWorkerRegistration,
+        showNotificationsTabOpen = true,
+        showNotificationsTabClosed = true,
+      } = config
+
+      onClickCallback && typeCheck("onClick", "function", onClickCallback)
+      typeCheck("showNotificationsTabOpen", "boolean", showNotificationsTabOpen)
+      typeCheck(
+        "showNotificationsTabClosed",
+        "boolean",
+        showNotificationsTabClosed,
+      )
+
+      if (!this._hasPermissionToSendNotifications()) {
+        return Promise.reject("Failed to get permission to send notifications")
+      }
+
+      const actions = []
+      if (showNotificationsTabOpen) {
+        actions.push(this._enableTabOpenNotifications())
+      }
+
+      if (showNotificationsTabClosed) {
+        actions.push(
+          this._enableTabClosedNotifications(serviceWorkerRegistration),
+        )
+      } else {
+        actions.push(this._disableTabClosedNotifications())
+      }
+
+      return Promise.all(actions).catch(err => {
+        this.logger.warn(`Chatkit error when enabling push notifications:`, err)
+        return Promise.reject(
+          `Chatkit error when enabling push notifications: ${err}`,
+        )
+      })
+    } catch (err) {
+      this.logger.warn(`Chatkit error when enabling push notifications:`, err)
+      return Promise.reject(
+        `Chatkit error when enabling push notifications: ${err}`,
+      )
+    }
+  }
+
+  disablePushNotifications() {
+    return this._disableTabClosedNotifications().then(() =>
+      this._disableTabOpenNotifications(),
+    )
+  }
+
+  _hasPermissionToSendNotifications() {
+    return Notification.requestPermission().then(
+      () => Notification.permission === "granted",
```

Comment:
you get handed the permission in the callback

```suggestion
      p => p === "granted"
```